services/diversity/src/services/richness.js

```diff
@@ -12,105 +12,112 @@ module.exports = (RichnessPersistence, restAPI) => {
+        });
     },
 
     /**
      * Get thresholds for the number of species for the given area type in the given group
      *
      * @param {String} areaType area type
+     * @param {String | Number} areaId area id
      * @param {String} group group to filter data (default to all), options are: 'all', 'total',
      * 'endemic', 'invasive', 'threatened'.
      *
      * @returns {Object[]} Number of inferred and observed species for the desired group.
      */
-    getNOSThresholds: async (areaType, group = 'all') => {
-      const data = [
-        {
-          id: 'total',
-          min_inferred: 10,
-          min_observed: 4,
-          max_inferred: 90,
-          max_observed: 85,
-        },
-        {
-          id: 'endemic',
-          min_inferred: 2,
-          min_observed: 4,
-          max_inferred: 75,
-          max_observed: 80,
-        },
-        {
-          id: 'invasive',
-          min_inferred: 3,
-          min_observed: 1,
-          max_inferred: 60,
-          max_observed: 65,
-        },
-        {
-          id: 'threatened',
-          min_inferred: 3,
-          min_observed: 5,
-          max_inferred: 50,
-          max_observed: 60,
-        },
-      ];
+    getNOSThresholds: async (areaType, areaId, group = 'all') => {
+      const promises = [];
       switch (group) {
         case 'total':
-          return [data[0]];
+          promises.unshift(
+            RichnessPersistence.findThresholdsTotalNumberOfSpecies(areaType, areaId),
+          );
+          break;
         case 'endemic':
-          return [data[1]];
+          promises.unshift(
+            RichnessPersistence.findThresholdsEndemicNumberOfSpecies(areaType, areaId),
+          );
+          break;
         case 'invasive':
-          return [data[2]];
+          promises.unshift(
+            RichnessPersistence.findThresholdsInvasiveNumberOfSpecies(areaType, areaId),
+          );
+          break;
         case 'threatened':
-          return [data[3]];
+          promises.unshift(
+            RichnessPersistence.findThresholdsThreatenedNumberOfSpecies(areaType, areaId),
+          );
+          break;
         case 'all':
+          promises.unshift(
+            RichnessPersistence.findThresholdsTotalNumberOfSpecies(areaType, areaId),
+            RichnessPersistence.findThresholdsEndemicNumberOfSpecies(areaType, areaId),
+            RichnessPersistence.findThresholdsInvasiveNumberOfSpecies(areaType, areaId),
+            RichnessPersistence.findThresholdsThreatenedNumberOfSpecies(areaType, areaId),
+          );
+          break;
         default:
-          return data;
+          throw new Error('Data doesn\'t exists for the given group');
       }
+      return Promise.all(promises)
+        .then((response) => {
+          if (group !== 'all') {
+            return response.map((item) => {
+              if (item.length > 0) return { id: group, ...item[0] };
```

Comment:
I think you need to change this condition, check what returns the query when there is no data
![1](https://user-images.githubusercontent.com/1734998/125860553-c4b1b004-43b7-46d4-94dd-66c1be0391cb.png)

Reply:
Yes indeed. That was the reason why in previous commit I changed to return an error.

Reply:
but you shouldn't return an error, but an empty array or at least an array of empty arrays as the other function

Reply:
I changed the logic inside .then() because always returned the ids (when values were nulls) despite of the new `condition.!item.every(element => element === null)`.